File: second_largest_number_in_an_array.js
function second_largest_number(arr){
    let first_max = -1
    let second_max= -1
    for (let i=0; i<arr.length; i++){
        if (arr[i]>first_max){
            second_max = first_max
            first_max = arr[i]
        }
        else if (arr[i]<first_max && arr[i]>second_max){
            second_max = arr[i]
        }
    }
    return second_max
}

let arr = [11,12,34,25,67,24,89]
result = second_largest_number(arr)
console.log(result)

File: second_largest_number_in_an_array.py
def second_largest_number(arr):
    if len(arr) < 2:
        return "Array must contain at least two elements."

    first_max = float('-inf')
    second_max = float('-inf')

    for num in arr:
        if num > first_max:
            second_max = first_max
            first_max = num
        elif second_max < num < first_max:
            second_max = num

    if second_max == float('-inf'):
        return "No second largest number found (all elements are equal)."

    return second_max

arr = [int(i) for i in input("enter numbers into the array: ").split()]
result = second_largest_number(arr)
print(result)